src/Parser.js

```diff
@@ -32,6 +34,9 @@ class Parser {
         if (this.constructor === Parser) {
             throw new Error("Cannot instantiate abstract class Plugin directly!");
         }
+        if (!options) return;
+        this.API = options.API;
```

Comment:
There is no definition of the `API` object interface in this package.

Additionally, I think it's unnecessary to store dependencies nested in `API` property.

Reply:
The logger is actually different based on the parameter to it..  `getLogger("my.unique.logging.id")`
Once the plugins call this.API.getLogger, it should store the results of that call.

I added the LintAPI object to document what the API is... it only includes the call back for the getLogger so far. Maybe we should just provide the getLogger function in the options object for the constructor? That way, the constructor for each plugin can call getLogger with its unique id to get its own logger instance, and then we don't need the API that way...